plugins/modules/purefa_pg.py

```diff
@@ -746,67 +998,46 @@ def update_pgroup(module, array):
     changed = True
     if not module.check_mode:
-        if ":" in module.params["name"]:
-            if "::" not in module.params["name"]:
-                try:
-                    target = "".join(module.params["target"])
-                    array.destroy_pgroup(
-                        module.params["name"], on=target, eradicate=True
-                    )
-                except Exception:
-                    module.fail_json(
-                        msg="Eradicating pgroup {0} failed.".format(
-                            module.params["name"]
-                        )
-                    )
-            else:
-                try:
-                    array.destroy_pgroup(module.params["name"], eradicate=True)
-                except Exception:
-                    module.fail_json(
-                        msg="Eradicating pgroup {0} failed.".format(
-                            module.params["name"]
-                        )
-                    )
+        if LooseVersion(RETENTION_LOCK_VERSION) <= LooseVersion(api_version):
+            res = array.delete_protection_groups(
+                names=[module.params["name"]], context_names=[module.params["context"]]
+            )
         else:
-            try:
-                array.destroy_pgroup(module.params["name"], eradicate=True)
-            except Exception:
-                module.fail_json(
-                    msg="Eradicating pgroup {0} failed.".format(module.params["name"])
+            res = array.delete_protection_groups(names=[module.params["name"]])
+        if res.status_code != 200:
+            module.fail_json(
+                msg="Eradicating pgroup {0} failed. Error: {1}".format(
+                    module.params["name"], res.errors[0].message
                 )
+            )
     module.exit_json(changed=changed)
 
 
 def delete_pgroup(module, array):
     """Delete Protection Group"""
+    api_version = array.get_rest_version()
     changed = True
     if not module.check_mode:
-        if ":" in module.params["name"]:
-            if "::" not in module.params["name"]:
-                try:
-                    target = "".join(module.params["target"])
-                    array.destroy_pgroup(module.params["name"], on=target)
-                except Exception:
-                    module.fail_json(
-                        msg="Deleting pgroup {0} failed.".format(module.params["name"])
-                    )
-            else:
-                try:
-                    array.destroy_pgroup(module.params["name"])
-                except Exception:
-                    module.fail_json(
-                        msg="Deleting pgroup {0} failed.".format(module.params["name"])
-                    )
+        if LooseVersion(RETENTION_LOCK_VERSION) <= LooseVersion(api_version):
+            res = array.patch_protection_groups(
+                names=[module.params["name"]],
+                context_names=[module.params["context"]],
```

Comment:
For non fleet array context_names param is causing trouble, outer IF loop is checking RETENTION_LOCK_VERSION, might need to relook at this.